src/utils/string.cpp

```diff
@@ -404,6 +404,100 @@ std::string toWide(const std::string &str, std::string::size_type length) {
 	return result;
 }
 
+/**
+* @brief Converts unicode @a bytes to ASCII string.
+*
+* @param[in] bytes Bytes for conversion.
+* @param[in] nBytes Number of bytes.
+*
+* @return Converted string in ASCII.
+*/
+std::string unicodeToAscii(const std::uint8_t *bytes, std::size_t nBytes)
+{
+	std::stringstream result;
+	if (!bytes || !nBytes)
+	{
+		return "";
+	}
+	if (nBytes & 1)
+	{
+		nBytes--;
+	}
+
+	for (std::size_t i = 0; i < nBytes; i += 2)
+	{
+		if (bytes[i] == 0 && bytes[i + 1] == 0)
+		{
+			break;
+		}
+		if (bytes[i + 1] == 0 && isPrintableChar(bytes[i]))
+		{
+			result << bytes[i];
+		}
+		else
+		{
+			const std::size_t maxC = std::pow(2, sizeof(std::string::value_type) * CHAR_BIT) - 1;
```

Comment:
If you are doing power of 2 then it's much better to use bit shift rather than `std::pow()`.